components/App/ArbSwap/Dashboard/PaymentMethod.tsx

```diff
@@ -1,35 +1,49 @@
 'use client'
 
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 
 import Button from '@/components/Common/Button'
 import Modal from '@/components/Common/Modal'
 import getFormattedDateTime from '@/hooks/useCurrentDate'
 import { modalStore } from '@/hooks/useStore'
+import redstone from "redstone-api";
 
 function PaymentMethod() {
   const modalKey = 'paymentMethod'
   const otherModalKey = 'pricing'
+  const [solToUSD, setSolToUSD] = useState(0.0);
 
   const { toggleModal } = modalStore()
 
   const handleNext = () => {
     toggleModal(modalKey)
     toggleModal(otherModalKey)
   }
+
+
   const handleClose = () => {
     toggleModal(modalKey)
   }
+
+  const fetchSolToUSD = async () => {
```

Comment:
If you are going to fetch the price in the same way from different places, try to create your own hook and use your custom hook in the places that you need its functionality.